File: Python/OOP/Encapsulation/Encapsulation (example 1).py
# Python does not have the private keyword, unlike some other object oriented languages.
# But encapsulation can be done with the following method (member can be private by double underscore)
class Student:
    def setName(self, studentName):
        self.__name = studentName
    
    def getName(self):
        print("Name: ", self.__name)

        
class encap:
    naiem = Student()
    naiem.setName("Abdullah Al Naiem")
    naiem.getName()

File: C++/OOP/Encapsulation/Encapsulation (example 1).cpp
#include <iostream>
using namespace std;

class Student{
    private:
        string name;
    
    public:
        void setName(string studentName){
            name = studentName;
        }

        void getName(){
            cout << "Name: " << name;
        }
};

int main(){
    Student naiem;
    naiem.setName("Abdullah Al Naiem");
    naiem.getName();
}